tests/test_workers/test_pkg_managers/test_gomod.py

```diff
@@ -239,6 +243,8 @@ def test_resolve_gomod_vendor_dependencies(
     mock_run.side_effect = run_side_effects
 
     mock_golang_version.return_value = "v0.1.0"
+    mock_go_release.return_value = "go0.1.0"
+    mock_get_gomod_version.return_value = "0.1.1"
```

Comment:
If we parametrized these values, we could test the algorithm that chooses the go binary that will be used to process the request (or maybe split this into a separate test?). The coveralls report is broken, but see the following uncovered lines in pkg_managers.gomod: L332-333, L341 and L354.

Reply:
trying to adjust the existing `resolve_gomod` test by parametrizing this and figuring out whether we used the right version would be non-trivial, so instead I extracted the logic to a new helper and tested the helper instead as suggested.